pkg/server/httpAPI.go

```diff
@@ -84,8 +108,11 @@ func (h *HTTPHandler) serveInitialize(rw http.ResponseWriter, req *http.Request)
 		h.initializationStatus = initializationStatusProgress
 		go func() {
 			// This is needed to stop snapshotter.
-			var s struct{}
-			h.StopCh <- s
+			h.AckStatusMutex.Lock()
```

Comment:
Same check and set func mentioned for the snapshotter might be helpful here as well?